pkg/controller/controller_instance.go

```diff
@@ -714,21 +714,28 @@ func (c *controller) pollServiceInstance(instance *v1beta1.ServiceInstance) erro
 		switch {
 		case deleting:
 			// For deprovisioning only, we should reattempt even on failure
-			//
-			// TODO(mkibbe): This is actually broken today, as we
-			// will not actually be retrying the deprovision, but
-			// instead constantly re-hitting the "last_operation"
-			// endpoint with the same operation, which will forever
-			// be stuck on failure.
 			msg := "Deprovision call failed: " + description
 			readyCond := newServiceInstanceReadyCondition(v1beta1.ConditionUnknown, errorDeprovisionCalledReason, msg)
 
 			if c.reconciliationRetryDurationExceeded(instance.Status.OperationStartTime) {
 				return c.processServiceInstancePollingFailureRetryTimeout(instance, readyCond)
 			}
 
-			c.processServiceInstanceOperationError(instance, readyCond)
-			return c.continuePollingServiceInstance(instance)
+			setServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionReady, readyCond.Status, readyCond.Reason, readyCond.Message)
```

Comment:
Logic seems a bit tortured to me, to create the condition up above just to decompose it here. 

Not sure there's anything to be done.

Reply:
I agree, but it's out of scope of this PR IMO. That one's my fault; I was hoping to eventually get to changing `setServiceInstanceCondition` to just take in the condition to set, but hadn't gotten to it yet. Should be a pretty easy swap job once that is done.